### ISSUE TITLE
Using AI in NLog breaks file / console loggers

### ISSUE THREAD

**Issue by mrtristan** [microsoft/ApplicationInsights-dotnet-logging#293]
i found two other issues where this is reported, but both people that opened them closed them without much info:
https://github.com/microsoft/ApplicationInsights-dotnet-logging/issues/137
https://github.com/microsoft/ApplicationInsights-dotnet-logging/issues/292

it seems that if the AI target is enabled, NLog console / file loggers cease working. If you simply comment out the AI target and rule, they start working again.

sample config:

```
<?xml version="1.0" encoding="utf-8"?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">

  <extensions>
    <add assembly="Microsoft.ApplicationInsights.NLogTarget" />
  </extensions>
  
  <targets async="true">
    <target xsi:type="File" name="logfile" fileName="D:\logs\${shortdate}.log" layout="${longdate} [${threadid}] ${uppercase:${level}} ${message}" />
    <target xsi:type="File" name="errorlogfile" fileName="D:\logs\error_${shortdate}.log" layout="${longdate} [${threadid}] ${uppercase:${level}} ${message}" />
    <target type="ApplicationInsightsTarget" name="aiTarget" />
  </targets>

  <rules>
    <logger name="*" minlevel="Info" writeTo="logfile" />
    <logger name="*" minlevel="Error" writeTo="errorlogfile" />
    <logger name="*" minlevel="Info" writeTo="aiTarget" />
  </rules>
</nlog>
```

**Comment by snakefoot** [comment]
Have you checked [NLog Logging Troubleshooting](https://github.com/NLog/NLog/wiki/Logging-troubleshooting) and looked at [NLog InternalLogging](https://github.com/NLog/NLog/wiki/Internal-Logging) ?

**Comment by mrtristan** [comment]
no because it's worked fine without app insights. turning on the app insights target is what causes it to not work. nlog on its own behaves just fine. and when app insights is enabled, i DO receive nlog messages in app insights. it seems that app insights is doing something that causes other non-ai targets to not work.

**Comment by Dmitry-Matveev** [comment]
In the other issues, it looked like after AI Package installation several files with the configuration were created. One took precedence and others were ignored. Merging the files helped with the issue.

@mrtristan , have you experienced the same? It means we may need to look at our Nuget package installation and avoid creating the second file.

**Comment by mrtristan** [comment]
@Dmitry-Matveev we manage our nuget configuration in a separate `nuget.config` (like above).

just for fun, i removed `Microsoft.ApplicationInsights.NLogTarget` from my project's packages.config and re-added it via the package manager. this yielded the following addition to my `web.config`

``` 
<nlog>
    <extensions>
      <add assembly="Microsoft.ApplicationInsights.NLogTarget" />
    </extensions>
    <targets>
      <target type="ApplicationInsightsTarget" name="aiTarget" />
    </targets>
    <rules>
      <logger name="*" minlevel="Trace" writeTo="aiTarget" />
    </rules>
  </nlog>
```

typically, i then delete this entry so that i can manage the configuration in the `nuget.config`. is this the potential duplication that you're describing? If so, i can confirm that only the `nuget.config` exists and not the `web.config` side.

**Comment by Dmitry-Matveev** [comment]
Yes, having two files (web.config and NLog config) that configured NLog to have AI target was a culprit in some of the references issues.

Your repro appears to be different and something we didn't see before. I'll add this investigation as a backlog item, however, judging by the current load on the team, I can't promise a quick resolution.

If the issue is more widespread, we'd appreciate more upvotes to help us prioritize accordingly.

**Comment by AVee** [comment]
> If the issue is more widespread, we'd appreciate more upvotes to help us prioritize accordingly.

I ran into this. I spotted it pretty quickly, but it was still unexpected.

I'm keeping the NLog config out of the web.config so I can auto-reload of the logging config without having to change my web.config (which would reload the entire application). I suspect I'm not the only one doing that. It also seems to be fairly common to fully configure logging in code.

In general, my expectation when installing a logging plugin would be that it's sitting there unconfigured, leaving it up to me to set it up the way I want. There's other choices to be made, perhaps I only want to log specific loggers, or use a different loglevel, etc.

**Comment by mrtristan** [comment]
@AVee to be clear, are you describing the issue where the non-AI loggers do not work when the AI target is enabled, or are you describing the auto-creation of configs that may conflict with the non-web.config file?

**Comment by AVee** [comment]
The latter, although in my case the automatic changes made to the web.config caused my other loggers (configured in web.nlog) to stop working. As such you could say installing the nuget package breaks other loggers in some scenarios. However, a quick test suggests the same happens if I add any other logger in the web.config, so I'm assuming this is just how NLog works (i.e. web.config overrules web.nlog).

Once the the changes in web.config are reverted and the ai-logger is moved into web.nlog with the rest the other loggers start working again. As such I don't see anything which suggest the ai-logging _code_ breaks something, for me it's just the default config being automatically installed which broke stuff. I can't speak for anyone else though ;-)

**Comment by r2d2rigo** [comment]
Chiming in because we have just discovered this very same issue - our server stopped logging any data to a file the same day we added Application Insights to a web app project.

**Comment by Dmitry-Matveev** [comment]
To summarize the current state, @AVee , @r2d2rigo most likely experience issue #266 (where two separate confgiurations for NLog are created, one by AI Nuget install in web.config, another by NLog beforehand), while @mrtristan got logging broken simply by enabling AI target in addition to File/Console manually as per instruction (and in one file).

We can open #266 back and discuss whether we think disabling config modification by nuget is the right way to address it (some would expect one-click onboarding where config is updated, some would expect they need to configure NLog targets manually).
